docs/practical-examples/forms.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: default
+title: Forms
+parent: Practical Examples
+---
+
+# Forms
+
+## Table of contents
+
+1. [Before you start](#before-you-start)
+2. [Using form helpers inside a `ViewComponent`](#using-form-helpers-inside-a-view-component)
+3. [Creating your own input elements](#creating-your-own-input-elements)
+    1. [Writing input elements in vanilla html](#writing-input-elements-in-vanilla-html)
+4. [Gotchas and known issues](#gotchas-and-known-issues)
+
+## Before you start
+It's important to understand the building blocks of forms in Ruby on Rails before attempting to use `ViewComponent` to enhance them.
+
+Please refer to the official documentation and Rails Guides to get up to speed.
+
+- `ActionView` form helpers [visit](https://guides.rubyonrails.org/form_helpers.html)
+- `ActionView::Helpers::FormBuilder` [visit](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/FormBuilder.html){:target="_blank"}
+
+After reading these, you should have a good understanding of:
+- The `form_with` helper used to create an HTML `<form>`
+- `FormBuilder` and the instance yielded to the `form_with` block
+- The different helper methods you can call on the `FormBuilder` instance to create input fields (e.g. `form.text_field`)
+- The conventions set by Rails to [name parameters](https://guides.rubyonrails.org/form_helpers.html#understanding-parameter-naming-conventions){:target="_blank"}
+
+## Using form helpers inside a `ViewComponent`
+
+If your component needs to render a form, you can use the `form_with` helpers just as you would in a Rails template or patial
+
+```ruby
+# app/components/comment_form_component.rb
+class CommentFormComponent < ApplicationComponent
+  def initialize(comment:)
+    @comment = comment
+  end
+  attr_reader :comment
+end
+```
+
+
+```erb
+<!-- app/components/comment_form_component.html.erb -->
+<%= form_with(model: comment) do |form| %>
+  <%= form.label :content %>
+  <%= form.text_area :content %>
+<% end %>
+```
+
+## Creating your own input elements
+
+Let's imagine you want to abstract the `label` and `text_area` into a single reusable class or method. Rails has done forms for quite a while now and they have a very stable API. Because of this, it's better to leverage the framework's conventions for extending the dsl available to create forms.
+
+### ⛔️ Bad: Passing the `FormBuilder` instance to a component
```

Comment:
I would ask "why?" reading this. Can we add some arguments here, or is the last paragraph enough?

Reply:
Thanks for pointing it out. After reading this with a fresh set of eyes I definitely saw an opportunity to enhance it. I restructured and rewrote this section. Let me know what you think!